app/src/main/java/com/github/se/studybuddies/MainActivity.kt

```diff
@@ -264,7 +264,9 @@ class MainActivity : ComponentActivity() {
                     backStackEntry ->
                   val groupUID = backStackEntry.arguments?.getString("groupUID")
                   if (groupUID != null) {
-                    SharedTimerScreen(navigationActions, SharedTimerViewModel(groupUID))
+                    val viewModel2 = SharedTimerViewModel.getInstance(groupUID)
```

Comment:
Maybe you could write SharedTimerViewModel.getInstance(groupUID) inside of the SharedTimerScreen ? I don't think it's so important but it saved from the creation of variable that is really used elsewhere

Reply:
I prefer having a variable because if in thee future I make changes it is better to change just the variable than all places where it is used